crates/anstyle-svg/src/lib.rs

```diff
@@ -504,20 +504,31 @@ fn write_fg_span(buffer: &mut String, span: SpanKind, element: &adapter::Element
         classes.push("hidden");
     }
 
-    let mut need_closing_a = false;
-
-    write!(buffer, r#"<{span}"#).unwrap();
-    write_classes(buffer, classes);
-    write!(buffer, r#">"#).unwrap();
-    if let Some(hyperlink) = &element.url {
-        write!(buffer, r#"<a href="{hyperlink}">"#).unwrap();
-        need_closing_a = true;
-    }
-    write!(buffer, "{fragment}").unwrap();
-    if need_closing_a {
-        write!(buffer, r#"</a>"#).unwrap();
+    if span != SpanKind::Tspan && classes.is_empty() && element.url.is_none() {
+        // No need to create an element if there is no class or href.
+        write!(buffer, "{fragment}").unwrap();
+    } else if span == SpanKind::Tspan {
+        write!(buffer, "<tspan").unwrap();
+        write_classes(buffer, classes);
+        buffer.write_str(">").unwrap();
+        if let Some(hyperlink) = &element.url {
+            write!(buffer, r#"<a href="{hyperlink}">{fragment}</a>"#).unwrap();
+        } else {
+            write!(buffer, "{fragment}").unwrap();
+        }
+        write!(buffer, "</tspan>").unwrap();
+    } else {
+        let closing = if let Some(hyperlink) = &element.url {
+            write!(buffer, r#"<a href="{hyperlink}"#).unwrap();
+            "</a>"
+        } else {
+            write!(buffer, "<span").unwrap();
+            "</span>"
+        };
+        write_classes(buffer, classes);
+        buffer.write_str(">").unwrap();
+        write!(buffer, "{fragment}{closing}").unwrap();
     }
```

Comment:
There are several changes still here
- No need to create an element
- changing how closing happens
- Splitting out tspan and span (should that be a `match`?)
- Applying classes onto the `a` if needed